app/controllers/api/v1/users_controller.rb

```diff
@@ -11,6 +11,20 @@ def index
         render json: users, status: :ok
       end
 
+      def destroy_self
+        authorize current_user, :destroy_self?
+
+        unless current_user.valid_password?(params[:password])
```

Comment:
this way we are having the logic at controller level, I know this is something simple, but I think the standard to this project is to have this kind of thing at a service class